### DISCUSSION TITLE
i3 Showing workspace buttons twice

### DISCUSSION THREAD

**Discussion by Tarinaky** [i3/i3#5902] [Q&A] [answered]
I tried to set the colors of my workspace buttons using this guide: https://www.schotty.com/Cheatsheets/Pywal_cheatsheet/

However, now my workspace buttons are appearing twice, once with the default colors, and once with the chosen colors, stacked on top of each other.
I assume I've done something wrong, but I can't see anyone else who's had this issue?

**Comment by orestisfl** [comment]
Can you please share your config file? `i3 --moreversion` should tell you where it is.

**Reply by Tarinaky** [reply]
font pango:monospace 8.000000
floating_modifier Mod1
default_border normal 2
default_floating_border normal 2
hide_edge_borders none
focus_wrapping yes
focus_follows_mouse no
focus_on_window_activation smart
mouse_warping output
workspace_layout default
workspace_auto_back_and_forth no
client.focused #4c7899 #285577 #ffffff #2e9ef4 #285577
client.focused_inactive #333333 #5f676a #ffffff #484e50 #5f676a
client.unfocused #333333 #222222 #888888 #292d2e #222222
client.urgent #2f343a #900000 #ffffff #900000 #900000
client.placeholder #000000 #0c0c0c #ffffff #000000 #0c0c0c
client.background #ffffff

bindsym Mod1+0 workspace number 10
bindsym Mod1+1 workspace number 1
bindsym Mod1+2 workspace number 2
bindsym Mod1+3 workspace number 3
bindsym Mod1+4 workspace number 4
bindsym Mod1+5 workspace number 5
bindsym Mod1+6 workspace number 6
bindsym Mod1+7 workspace number 7
bindsym Mod1+8 workspace number 8
bindsym Mod1+9 workspace number 9
bindsym Mod1+Down focus down
bindsym Mod1+Left focus left
bindsym Mod1+Return exec i3-sensible-terminal
bindsym Mod1+Right focus right
bindsym Mod1+Shift+0 move container to workspace number 10
bindsym Mod1+Shift+1 move container to workspace number 1
bindsym Mod1+Shift+2 move container to workspace number 2
bindsym Mod1+Shift+3 move container to workspace number 3
bindsym Mod1+Shift+4 move container to workspace number 4
bindsym Mod1+Shift+5 move container to workspace number 5
bindsym Mod1+Shift+6 move container to workspace number 6
bindsym Mod1+Shift+7 move container to workspace number 7
bindsym Mod1+Shift+8 move container to workspace number 8
bindsym Mod1+Shift+9 move container to workspace number 9
bindsym Mod1+Shift+Down move down
bindsym Mod1+Shift+Left move left
bindsym Mod1+Shift+Right move right
bindsym Mod1+Shift+Up move up
bindsym Mod1+Shift+c reload
bindsym Mod1+Shift+e exec i3-nagbar -t warning -m 'Do you want to exit i3?' -b 'Yes' 'i3-msg exit'
bindsym Mod1+Shift+minus move scratchpad
bindsym Mod1+Shift+q kill
bindsym Mod1+Shift+r restart
bindsym Mod1+Shift+space floating toggle
bindsym Mod1+Up focus up
bindsym Mod1+a focus parent
bindsym Mod1+d exec rofi -show combi -show-icons -window-thumbnail
bindsym Mod1+e layout toggle split
bindsym Mod1+f fullscreen toggle
bindsym Mod1+h split h
bindsym Mod1+minus scratchpad show
bindsym Mod1+r mode resize
bindsym Mod1+s layout stacking
bindsym Mod1+space focus mode_toggle
bindsym Mod1+v split v
bindsym Mod1+w layout tabbed

mode "resize" {
  bindsym Down resize grow height 10 px or 10 ppt
  bindsym Escape mode default
  bindsym Left resize shrink width 10 px or 10 ppt
  bindsym Return mode default
  bindsym Right resize grow width 10 px or 10 ppt
  bindsym Up resize shrink height 10 px or 10 ppt
}

bar {
  font pango:monospace 8.000000
  mode dock
  hidden_state hide
  position bottom
  status_command /nix/store/sh0yr7c1wd6klpx3q513ykn3bc8c2ndv-i3status-2.14/bin/i3status
  i3bar_command /nix/store/jxh6dy636d590j9rjbxp2ydar06b6700-i3-4.23/bin/i3bar
  workspace_buttons yes
  strip_workspace_numbers no
  tray_output primary
  colors {
    background #000000
    statusline #ffffff
    separator #666666
    focused_workspace #4c7899 #285577 #ffffff
    active_workspace #333333 #5f676a #ffffff
    inactive_workspace #333333 #222222 #888888
    urgent_workspace #2f343a #900000 #ffffff
    binding_mode #2f343a #900000 #ffffff
  }
}

gaps inner 4
smart_gaps on
smart_borders on
exec  $TERMINAL
exec  firefox
exec  blueman-applet
exec  pa-applet
 assign [class="^firefox$"] workspace 1
 assign [class="^obsidian$"] workspace 2
 assign [class="^Code$"] workspace 2

set_from_resource $fg i3wm.color7 #f0f0f0
set_from_resource $bg i3wm.color2 #f0f0f0
set_from_resource $hi i3wm.color4 #f0f0f0
client.focused $bg $hi #000000 $bg $bg
client.focused_inactive $bg $bg $fg $bg $bg
client.unfocused $bg $bg $fg $bg $bg
client.urgent $bg $bg $fg $bg $bg
client.placeholder $bg $bg $fg $bg $bg
client.background $bg 

bar {
 colors {
  focused_workspace $bg $bg #000000
  active_workspace $fg $fg #000000
  inactive_workspace $fg $fg #000000
 }
 
 }

**Reply by Tarinaky** [reply, marked as answer]
I now see the problem. Thanks for your help *facepalm*

There's two bar{} blocks.

**Reply by orestisfl** [reply]
You only want one `bar{}` section, each new `bar{}` section creates a new i3bar.

Move the
```
 colors {
  focused_workspace $bg $bg #000000
  active_workspace $fg $fg #000000
  inactive_workspace $fg $fg #000000
 }
```
part inside
```
bar {
font pango:monospace 8.000000
mode dock
hidden_state hide
position bottom
status_command /nix/store/sh0yr7c1wd6klpx3q513ykn3bc8c2ndv-i3status-2.14/bin/i3status
i3bar_command /nix/store/jxh6dy636d590j9rjbxp2ydar06b6700-i3-4.23/bin/i3bar
workspace_buttons yes
strip_workspace_numbers no
tray_output primary
colors {
background #000000
statusline #ffffff
separator #666666
focused_workspace #4c7899 #285577 #ffffff
active_workspace #333333 #5f676a #ffffff
inactive_workspace #333333 #222222 #888888
urgent_workspace #2f343a #900000 #ffffff
binding_mode #2f343a #900000 #ffffff
}
}
```
replacing duplicate colors

**Reply by Tarinaky** [reply]
Further information, for anyone who comes after me: There doesn't actually seem to be a sensible way to do it in Nix (my distro). The only way to set the colors involves blowing away the entire default config for every other attribute of the bar. This is very dumb.

**Reply by orestisfl** [reply]
I don't exactly understand the problem, even though I am missing some specifics in Nix.

> The only way to set the colors involves blowing away the entire default config for every other attribute of the bar. This is very dumb.

It seems that you can freely change the config as you want, given the file you pasted above. You don't need to "blow away" other parts of the bar. Perhaps a more specific example could help us fix whatever problem you are having?
